builder-support/helpers/install_quiche.sh

```diff
@@ -0,0 +1,30 @@
+#!/bin/sh
+set -v
+set -e
+
+readonly QUICHE_VERSION='0.18.0'
+readonly QUICHE_TARBALL="${QUICHE_VERSION}.tar.gz"
+readonly QUICHE_TARBALL_URL="https://github.com/cloudflare/quiche/archive/${QUICHE_TARBALL}"
+readonly QUICHE_TARBALL_HASH='eb242a14c4d801a90b57b6021dd29f7a62099f3a4d7a7ba889e105f8328e6c1f'
+
+cd /tmp
+echo $0: Downloading $QUICHE_TARBALL
+curl -L -o "${QUICHE_TARBALL}" "${QUICHE_TARBALL_URL}"
+# Line below should echo two spaces between digest and name
+echo "${QUICHE_TARBALL_HASH}"  "${QUICHE_TARBALL}" | sha256sum -c -
+tar xf "${QUICHE_TARBALL}"
+cd "quiche-${QUICHE_VERSION}"
+RUST_BACKTRACE=1 cargo build --release --no-default-features --features ffi,boringssl-boring-crate --package quiche
+install -Dm644 quiche/include/quiche.h /usr/include
```

Comment:
On MacOS the build succeeds, but the install commands go wrong: `-D` is unknown, the target dir does not exists by default (it's even dependent on Intel vs ARM) and the shared lib name ends with `.dylib`.

Given that, we might just refuse to run on ! Linux.

Reply:
None of that is very hard to fix, I'll see if I can get access to a macOS env to test. A macOS runner for our CI, perhaps?

Reply:
I can give it a shot as well

Reply:
https://github.com/omoerbeek/pdns/tree/quiche-macos works for me on macOS

Reply:
Pushed, thanks!